src/SFA.DAS.ApprenticeCommitments.Web/Pages/Apprenticeships/ConfirmYourTrainingProvider.cshtml.cs

```diff
@@ -1,15 +1,54 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using SFA.DAS.ApprenticeCommitments.Web.Pages.IdentityHashing;
+using SFA.DAS.ApprenticeCommitments.Web.Services.OuterApi;
+using System.Threading.Tasks;
 
 namespace SFA.DAS.ApprenticeCommitments.Web.Pages.Apprenticeships
 {
-    public class ConfirmYourTrainingDetailsModel : PageModel
+    public class ConfirmYourTrainingModel : PageModel
     {
+        private readonly IOuterApiClient _client;
+        private readonly AuthenticatedUser _authenticatedUser;
+
         [BindProperty(SupportsGet = true)]
-        public string? ApprenticeshipId { get; set; }
+        public HashedId ApprenticeshipId { get; set; }
+
+        [BindProperty]
+        public string TrainingProviderName { get; set; }
+
+        [BindProperty]
+        public bool? ConfirmTrainingProvider { get; set; }
+
+        public string Backlink => $"/apprenticeships/{ApprenticeshipId.Hashed}";
+
+        public ConfirmYourTrainingModel(IOuterApiClient client, AuthenticatedUser authenticatedUser)
+        {
+            _authenticatedUser = authenticatedUser;
+            _client = client;
+        }
+
+        public async Task OnGetAsync()
+        {
+            var apprenticeship = await _client
+                .GetApprenticeship(_authenticatedUser.RegistrationId, ApprenticeshipId.Id);
+            TrainingProviderName = apprenticeship.TrainingProviderName;
+        }
 
-        public void OnGet()
+        public IActionResult OnPost()
         {
+            switch (ConfirmTrainingProvider)
+            {
+                case null:
+                    ModelState.AddModelError(nameof(ConfirmTrainingProvider), "Select an answer");
```

Comment:
I'm wondering if the content people have expressed an opinion on what this message should be. I like the simple message as we could use it all the the confirm pages. but from experience the content providers trend to be more explicit about teh message here. Don't think it was in the AT criteria

Reply:
That did come from CS-264 - AC4:
> Then display a red validation error message above the CTA with the text: “Select an answer” 

I think it's the same message in ConfirmYouEmployer?